code/controllers/DMSDocumentCartCheckoutPage_Controller.php

```diff
@@ -0,0 +1,223 @@
+     *
+     * To extend use the following from within an Extension subclass:
+     *
+     * <code>
+     * public function updateSend($email)
+     * {
+     *     // Do something here
+     * }
+     * </code>
+     * @return mixed
+     *
+     * @throws DMSDocumentCartException
+     */
+    public function send()
+    {
+        $member = $this->CartEmailRecipient();
+        if ($member->exists()) {
+            $cart = $this->getCart();
+            $from = Config::inst()->get('Email', 'admin_email');
+            $emailAddress = ($info = $cart->getReceiverInfo()) ? $info['ReceiverEmail'] : $from;
+            $email = Email::create(
+                $from,
+                $emailAddress,
+                _t('DMSDocumentCartCheckoutPage.EMAIL_SUBJECT', 'Request for Printed Publications')
+            );
+            $email->setBcc($member->Email);
+            $renderedCart = $cart->renderWith('DocumentCart_email');
+            $body = _t(
+                'DMSDocumentCartCheckoutPage.EMAIL_BODY',
+                '<p>A request for printed publications has been submitted with the following details:</p>'
+            );
+            $body .= $renderedCart->getValue();
+            $email->setBody($body)->setReplyTo($emailAddress);
+            $this->extend('updateSend', $email);
+
+            return $email->send();
+        } else {
+            throw new DMSDocumentCartException('No recipient has been configured. Please do so from the CMS');
+        }
+    }
+
+    /**
+     * Handles form submission.
+     * Totals requested are updated, delivery details added, email sent for fulfillment
+     * and print request totals updated.
+     *
+     * @param array          $data
+     * @param Form           $form
+     * @param SS_HTTPRequest $request
+     *
+     * @return SS_HTTPResponse
+     */
+    public function doRequestSend($data, Form $form, SS_HTTPRequest $request)
+    {
+        $this->updateCartItems($data);
+        $this->updateCartReceiverInfo($data);
+        $this->send();
+        $this->trackTimestampedPrintRequest();
+        $this->getCart()->saveSubmission($form);
+        $this->getCart()->emptyCart();
+
+        return $this->redirect($this->Link('complete'));
+    }
+
+    /**
+     * Displays the preconfigured thank you message to the user upon completion
+     *
+     * @return ViewableData_Customised
+     */
+    public function complete()
+    {
+        return $this->customise(
+            ArrayData::create(
+                array(
+                    'Content' => $this->ThanksMessage,
+                )
+            )
+        );
+    }
+
+    /**
+     * Increments the print counts of all documents which were successfully sent.
+     */
+    public function trackTimestampedPrintRequest()
+    {
+        /** @var DMSRequestItem $item */
+        foreach ($this->getCart()->getItems() as $item) {
+            /** @var DMSDocument|DMSDocumentCartExtension $doc */
```

Comment:
We've discussed this!

Also below, may as well just be `$item->getDocument()->incrementPrintRequest();` instead of assigning to a variable etc